src/inference/graph/tools/adapter_tools.py

```
@@ -0,0 +1,65 @@
+"""어댑터별 독립 도구를 동적으로 생성하는 팩토리 모듈.
+
+AdapterRegistry에 등록된 모든 어댑터를 순회하여
+각각에 대응하는 StructuredTool을 생성한다.
+adapters.yaml에 어댑터를 추가하면 자동으로 새 도구가 등록된다.
+"""
+
+from __future__ import annotations
+
+import json
+from typing import Callable, List
```

Ruff (0.15.9)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)